osu.Server.Queues.ScoreStatisticsProcessor/Commands/Score/ChangeModMultiplierCommand.cs

```diff
@@ -0,0 +1,163 @@
+
+        [Option(CommandOptionType.SingleValue, Template = "--start-id", Description = "The ID of the `scores` table row to start processing from.")]
+        public ulong? StartId { get; set; }
+
+        [Option(CommandOptionType.SingleOrNoValue, Template = "--batch-size", Description = "The maximum number of scores to fetch in each batch.")]
+        public int BatchSize { get; set; } = 5000;
+
+        [Option(CommandOptionType.SingleOrNoValue, Template = "--dry-run", Description = "Do not actually change any score totals, just display what would be done.")]
+        public bool DryRun { get; set; }
+
+        private readonly StringBuilder sqlBuffer = new StringBuilder();
+        private readonly ElasticQueuePusher elasticQueueProcessor = new ElasticQueuePusher();
+        private readonly HashSet<ElasticQueuePusher.ElasticScoreItem> elasticItems = new HashSet<ElasticQueuePusher.ElasticScoreItem>();
+
+        [UsedImplicitly]
+        public async Task<int> OnExecuteAsync(CommandLineApplication app, CancellationToken cancellationToken)
+        {
+            if (RulesetId == null
+                || string.IsNullOrEmpty(ModAcronym)
+                || OldMultiplier == null
+                || NewMultiplier == null)
+            {
+                await Console.Error.WriteLineAsync("One or more required parameters is missing.");
+                app.ShowHelp(false);
+                return 1;
+            }
+
+            if (NewMultiplier.Value == OldMultiplier.Value)
+            {
+                Console.WriteLine("New and old multipliers are equal - there is nothing to do.");
+                return 0;
+            }
+
+            ArgumentOutOfRangeException.ThrowIfNegativeOrZero(OldMultiplier.Value, nameof(OldMultiplier));
+            ArgumentOutOfRangeException.ThrowIfNegativeOrZero(NewMultiplier.Value, nameof(NewMultiplier));
+
+            Console.WriteLine();
+            Console.WriteLine($"Changing multiplier of mod {ModAcronym} in ruleset {RulesetId} from {OldMultiplier} to {NewMultiplier}");
+            Console.WriteLine($"Indexing to elasticsearch queue(s) {elasticQueueProcessor.ActiveQueues}");
+
+            if (DryRun)
+                Console.WriteLine("RUNNING IN DRY RUN MODE.");
+
+            Thread.Sleep(5000);
+
+            ulong lastId = StartId ?? 0;
+            int converted = 0;
+            int skipped = 0;
+
+            using var conn = DatabaseAccess.GetConnection();
+
+            while (!cancellationToken.IsCancellationRequested)
+            {
+                var scoresToAdjust = (await conn.QueryAsync<ScoreToAdjust>(
+                    "SELECT `id`, `total_score` "
+                    + "FROM `scores` "
+                    + "WHERE `id` BETWEEN @lastId AND (@lastId + @batchSize - 1) "
+                    + "AND `ruleset_id` = @rulesetId "
+                    + "AND JSON_SEARCH(`data`, 'one', @modAcronym, NULL, '$.mods[*].acronym') IS NOT NULL",

```

Comment:
Wasn't sure whether to do this sql-side or c#-side. I *think* this should be *somewhat* painless because `EXPLAIN` pegs this as a where filter over a range PK lookup so it shouldn't be _that_ slow.